examples/auth/auth_service.exs

```diff
@@ -0,0 +1,95 @@
+# Run this script from the Spell root with
+#
+#     mix spell.example.auth
+
+defmodule Auth do
+  @moduledoc """
+  The `RPC` module implements the example for asynchronous Remote Procedure Call.
+  """
+
+  defmodule Callee do
+    require Logger
+    defstruct [:callee, :procedure, timeout: 1000]
+
+    # Public Functions
+
+    @doc """
+    Start the callee with the passed procedure.
+    """
+    def start_link(topics, options \\ []) do
+      {:ok, spawn_link(fn -> init(topics, options) end)}
+    end
+
+    # Private Functions
+
+    defp init(procedure, options) do
+      {:ok, callee} = Auth.new_peer([Spell.Role.Callee], options)
+      {:ok, _register_id} = Spell.cast_register(callee, procedure)
+
+      %__MODULE__{callee: callee,
+                  procedure: procedure}
+        |> struct(options)
+        |> loop()
+    end
+
+    defp loop(state) do
+      receive do
+        :stop -> :ok
+        {Spell.Peer, pid, message} -> handle_message(state, message)
+        _ -> {:error, :wrong_message}
```

Comment:
The classic catch-all -- what you're doing here totally works, but you should ask whether the process's normal behavior includes receiving unexpected messages. It's sometimes better to have your code fail when it runs into an unexpected message because it's unexpected behavior